vending-app/src/main/java/org/microg/vending/ui/VendingActivity.kt

```diff
@@ -97,7 +98,22 @@ class VendingActivity : ComponentActivity() {
                     val client = HttpClient()
 
                     // Get download links for requested package
-                    val downloadUrls = runCatching { client.requestDownloadUrls(
+                    val downloadUrls = runCatching {
+
+                        // Purchase app (only needs to be done once, in theory)
+                        val parameters = mapOf(
+                            "ot" to "1",
+                            "doc" to app.packageName,
+                            "vc" to app.versionCode.toString()
+                        )
+                        client.post(
+                            url = URL_PURCHASE,
+                            headers = buildRequestHeaders(auth!!.authToken, auth!!.gsfId.toLong(16)),
+                            params = parameters,
+                            adapter = GoogleApiResponse.ADAPTER
+                        )
```

Comment:
@fynngodau
I don't know the code enough but does it check the microG option before purchase?

Reply:
That wouldn't make sense to me, as this is neither in-app billing nor during an app license check.

Reply:
I mean the option: Auto purchase free apps

Reply:
@fynngodau
It can also be a new option but it shouldn't automatically purchase apps (even if free) if not configured in microG settings.

Reply:
@fynngodau
If the existing option doesn't apply, I think we should add a new one, since anything that causes contact with a remote server (and thus potentially reduces privacy) should be able to be disabled by the user.